framework/db/src/com/cloud/utils/db/Transaction.java

```diff
@@ -35,18 +35,15 @@
         if (currentTxn != null) {
             databaseId = currentTxn.getDatabaseId();
         }
-        TransactionLegacy txn = TransactionLegacy.open(name, databaseId, false);
-        try {
-//            if (txn.dbTxnStarted()){
-//                String warnMsg = "Potential Wrong Usage: TRANSACTION.EXECUTE IS WRAPPED INSIDE ANOTHER DB TRANSACTION!";
-//                s_logger.warn(warnMsg, new CloudRuntimeException(warnMsg));
-//            }
+        try (final TransactionLegacy txn = TransactionLegacy.open(name, databaseId, false)) {
+            // if (txn.dbTxnStarted()){
+            // String warnMsg = "Potential Wrong Usage: TRANSACTION.EXECUTE IS WRAPPED INSIDE ANOTHER DB TRANSACTION!";
+            // s_logger.warn(warnMsg, new CloudRuntimeException(warnMsg));
+            // }
```

Comment:
Please remove crufty commented code.  git history/bisect is a far better tool to understand the evolution of code.